### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by JimmyWxzzz**

### CODE REVIEW

File: src/main/java/com/sweepy/controller/controller.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+@RestController
+public class controller {
+
+    private urlService urlService;
+
+    public controller (urlService urlService) {
+        this.urlService = urlService;
+    }
+    @GetMapping("/longToShort")
```
Comment (LeonYanc) [inline src/main/java/com/sweepy/controller/controller.java]: Should be POST

File: src/main/java/com/sweepy/service/urlServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        if (Objects.equals(method, "base62")) {
+            shortUrl = base62Converter.encode(longUrl);
+        } else {
+            shortUrl = randomConverter.encode(longUrl);
+        }
+
+        urlTable entry = new urlTable(longUrl, shortUrl, method);
+        urlRepository.save(entry);
+        return "http://www.sweepy.com/" + shortUrl;
```
Comment (LeonYanc) [inline src/main/java/com/sweepy/service/urlServiceImpl.java]: This prefix should be in the property

File: src/main/java/com/sweepy/converter/randomConverter.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        while (map.containsKey(key)) {
+            key = r.nextInt(Integer.MAX_VALUE);
+        }
+        map.put(key, longUrl);
+        return "" + key;
+    }
+
+    public String decode(String shortUrl) {
+        return map.get(Integer.parseInt(shortUrl.replace("http://www.sweepy.com/", "")));
+    }
```
Comment (zhangxyleon) [inline src/main/java/com/sweepy/converter/randomConverter.java]: should use the database to replace the In-Memory map

**Comment by JimmyWxzzz** [comment]
image

long url to short url using base62

image

long url to short url using random

image

short url to long url with base62

image

short url to long url with random

File: src/main/java/com/sweepy/converter/randomConverter.java (hunk 3 above)
Comment (JimmyWxzzz) [reply src/main/java/com/sweepy/converter/randomConverter.java]: I didn't update github but this method is removed therefore fixed.

File: src/main/java/com/sweepy/controller/controller.java (hunk 1 above)
Comment (JimmyWxzzz) [reply src/main/java/com/sweepy/controller/controller.java]: changed

File: src/main/java/com/sweepy/service/urlServiceImpl.java (hunk 2 above)
Comment (JimmyWxzzz) [reply src/main/java/com/sweepy/service/urlServiceImpl.java]: fixed

**Review by LeonYanc — APPROVED** [review]

**Pull request merged**